infra/storage/kFile.class.php

```diff
@@ -722,6 +722,26 @@ public static function safeFilePutContents($filePath, $var, $mode=null)
 		}
 		return true;
 	}
+
+	/**
+	 * Lazy saving of file content to a temporary path, the file will exist in this location until the temp files are purged
+	 * @param string $fileContent
+	 * @param string $prefix
+	 * @param integer $permission
+	 * @param boolean $strict - check if to write file without content
+	 * @return string path to temporary file location
+	 */
+	public static function getTempFileWithContent($fileContent, $prefix = '' , $permission = null, $strict = false )
```

Comment:
From what I can see from the changes it would be mush better to set strict to true and pass false only when you don't want to validate the file content.